Jenkinsfile

```diff
@@ -0,0 +1,74 @@
+def dockerImage = "maven:3-jdk-8"
+def dockerArgs(env) { return "-e JAVA_TOOL_OPTIONS=-Duser.home=${env.WORKSPACE}"; }
+
+pipeline {
+    agent any
+
+    triggers {
+        cron('H H(20-23) * * 1-5')
+	pollSCM('H * * * 1-5')
+    }
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '5'))
+        timeout time: 60, unit: 'MINUTES'
+    }
+
+    environment {
+	SONAR_TOKEN=credentials('sonar-token')
+	ARTIFACTORY_DEPLOY=credentials('artifactory-deploy')
+    }
+    stages {
+        stage('Build') {
+            agent {
+                docker {
+		    image dockerImage
+		    reuseNode true
+		    args dockerArgs(env)
+                }
+            }
+            steps {
+		sh 'mvn -B clean package -Dmaven.test.failure.ignore'
```

Comment:
Is the `maven.test.failure.ignore` just a temporary flag? Or was it propagated from the current build definition?

Reply:
This is the way I found to distinguish between failing and unstable (tests failing) builds. The non-pipeline Jenkins Maven projects do this automatically somehow, here we let the maven step succeed and then the junit step will mark the build as unstable. We may not actually need this failing/unstable distinction, but Jenkins has it since forever.

Maybe https://plugins.jenkins.io/pipeline-maven/ would handle this automatically, but it is not installed and there are posts (eg. on stackoverflow) that suggests that people still need to use maven.test.failure.ignore with this plugin.